tests/trantest.h

```diff
@@ -548,10 +615,100 @@ trantest_mqtt_sub_pub(trantest *tt)
 		params.data_len = strlen(data);
 		params.qos      = qos;
 
-		transtest_mqtt_sub_send(tt->reqsock, &client);
+		client = nng_mqtt_client_alloc(tt->reqsock, &send_callback, true);
+		transtest_mqtt_sub_send(tt->reqsock, &client, true);
 		nng_msleep(200);
 		trantest_mqtt_pub(tt->repsock);
 		transtest_mqtt_sub_recv(tt->reqsock, &client);
+		transtest_mqtt_unsub_send(tt->reqsock, &client, true);
+		nng_mqtt_client_free(client, true);
+
+	});
+}
+
+void
+server_cb(void *arg)
+{
+	// printf("AIO START\n");
+}
+
+void
+transtest_broker_start(trantest *tt, nng_listener listener)
+{
+	conf *nanomq_conf;
+	nanomq_conf = nng_zalloc(sizeof(conf));
+	conf_init(nanomq_conf);
+	tt->repsock.data = nanomq_conf;
+	So(nng_nmq_tcp0_open(&tt->repsock) == 0);
+	nng_listener_create(&listener, tt->repsock, tt->addr);
+	nng_listener_set(listener, NANO_CONF, nanomq_conf, sizeof(conf));
+	if (nng_listener_start(listener, 0) != 0) {
+		nng_listener_close(listener);
+		return;
+	}
+	// alloc and init work
+	if ((work = nng_alloc(sizeof(*work))) == NULL) {
+	}
+	if (nng_aio_alloc(&work->aio, server_cb, work) != 0) {
+	}
+	if (nng_ctx_open(&work->ctx, tt->repsock) != 0) {
+		printf("ctx_open fail\n");
+	}
```

Comment:
Using assert or something else instead of IF or PRINTF ?

Reply:
got it